### ISSUE TITLE
Too many unique verts, max = 65536 (map has too much brush geometry)

### ISSUE THREAD

**Issue by gooball60** [Facepunch/garrysmod-issues#5066]
I noticed this in the recent patch notes:

> Hammer: Increased MAX_MAP_OVERLAYS to 1024 (2x increase), increased MAX_MAP_BRUSHSIDES to a lot (~163k, ~3x increase), matching CS:GO limits

I thought, "at last, I'm free of those damned Source engine limits, I can finally make some more complex maps". Unfortunately it was too early to celebrate, as I ran into something I'd never seen before when compiling:

`Too many unique verts, max = 65536 (map has too much brush geometry)`

It shows up right after FixTjuncs in vbsp. Of course I could simply water down a lot of the map details, but my thinking is that the increased brushside limit means nothing if I'm running into this error instead. Might it be possible to increase this "unique verts" limit as well?

**Comment by jorjic** [comment]
> Might it be possible to increase this "unique verts" limit as well?

Nope. [Various data structures](https://github.com/ValveSoftware/source-sdk-2013/blob/master/mp/src/public/bspfile.h#L673-L677) used for the BSP format have arbitrarily capped vertex indices at 16-bits (2^16 = 65536). 

The vbsp, vvis, and vrad sources are available in the [Source SDK 2013 base](https://github.com/ValveSoftware/source-sdk-2013) and anyone can make changes and use the custom versions to compile maps. Increasing the limits will not necessarily mean that the map can be compiled that way, or have it work in the game. Some limits require breaking changes to the map format, and `MAX_MAP_VERTS` is one of them.
